### PR TITLE
Py3, Code Refactor, New Options

### PR DESCRIPTION
**Pull request by ghost**
Hello Pierre!

It didn't feel right to see Droopy go without updates in a few years, so this was my weekend project. :)

I've made Droopy run in Python 2.7 and 3.4, and refactored a lot of code. Changes I set out to do:
- Update to Python3, but keep Py2 because OpenWRT still supports legacy Python by default (e.g. PirateBox).
- Remove the global variable logic and replace it with scoped variables or class-variables where local scoping wasn't possible. This got messy at times, but never illegible I think.
- Replace opt parsing with argparse.
- Rearrange code to keep core logic at the top, data in the middle, and command-line logic at the bottom.
- Keep backward compatibility, but I can't test everything!

If I've done my job well (snort) this would be a drop-in replacement for Droopy in Piratebox etcetera, with downstream users not noticing the difference. But, it should be a bit more future proof and maintainable now.

Hope you like it!

### CODE REVIEW

**Comment by cathalgarvey** [comment]
Hey again; just wondering whether there's interest in this, because if not I'm happy to rename my fork "Droopy3" and strip out the Py2. What do you think?

**Comment by stackp** [comment]
Hey Cathal, thank you for your contribution and sorry for replying late! I'm definitely interested in this. It looks like a solid contribution and I will look into it ASAP.

**Comment by cathalgarvey** [comment]
Take your time, just wondering whether you were still around these parts! Glad you like it. :)

**Comment by stackp** [comment]
Thanks Cathal for your rock solid contribution, definitely a new youth for Droopy.

**Comment by cathalgarvey** [comment]
Thanks so much, proud to have this accepted! :)

**Comment by codybak4** [comment]
@stackp

File: Readme.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Droopy
+### Easy File Sharing
+Copyright 2008-2013 (c) Pierre Duquesne <stackp@online.fr>
```
Comment (codybak4) [inline Readme.md:3]:
```suggestion
Copyright 2008-2013 (c) Pierre Duquesne <stackp@online.fr>@copyright by @codybak4 
```

**Pull request merged**